crates/shim/src/lib.rs

```diff
@@ -290,22 +294,42 @@ fn handle_signals(mut signals: Signals) {
                     debug!("received {}", sig);
                     return;
                 }
-                SIGCHLD => loop {
-                    unsafe {
-                        let pid: pid_t = -1;
-                        let mut status: c_int = 0;
-                        let options: c_int = libc::WNOHANG;
-                        let res_pid = libc::waitpid(pid, &mut status, options);
-                        let status = libc::WEXITSTATUS(status);
-                        if res_pid <= 0 {
-                            break;
-                        } else {
-                            monitor_notify_by_pid(res_pid, status).unwrap_or_else(|e| {
-                                error!("failed to send exit event {}", e);
-                            });
+                SIGCHLD => {
```

Comment:
A little confused here.
The new implementation spawns a new working thread for each received `SIGCHLD` signal, and the working thread never exits. It has different behavior with original implementation.

Reply:
When pid is -1, it waits for any child process to exit. Because a working thread is spawn to handle the signal, new SIGCHLD signal may get delivered concurrently to the working thread. That means we may spawn multiple working threads, and all working threads will be kept alive.

Reply:
I see, thank you! 
Fixed.